File: backend/src/main/java/com/devmatch/backend/domain/project/dto/ProjectCreateRequest.java
package com.devmatch.backend.domain.project.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record ProjectCreateRequest(
    @NotNull @Size(min = 1, max = 200) String title,
    @NotNull @Size(min = 1, max = 2000) String description,
    @NotNull @Size(min = 1, max = 500) String techStack,
    @Min(1) int teamSize,
    @Min(1) int durationWeeks
) {

}

File: backend/src/main/java/com/devmatch/backend/domain/project/dto/ProjectCreateRequest.kt
package com.devmatch.backend.domain.project.dto

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.Size

data class ProjectCreateRequest(
    @field:Size(min = 1, max = 200)
    val title: String,
    @field:Size(min = 1, max = 2000)
    val description: String,
    @field:Size(min = 1, max = 500)
    val techStack: String,
    @field:Min(1)
    val teamSize: Int,
    @field:Min(1)
    val durationWeeks: Int
) 